pkgs/applications/virtualization/singularity/generic.nix

```diff
@@ -96,6 +96,11 @@ let
     ln -s ${lib.escapeShellArg newuidmapPath} "$out/bin/newuidmap"
     ln -s ${lib.escapeShellArg newgidmapPath} "$out/bin/newgidmap"
   '');
+  # Singularity implements CDI support since 4.0.0
+  # (https://github.com/sylabs/singularity/releases/tag/v4.0.0-rc.1). Before
+  # this version, we need to use the runtime wrapper to enable GPU
+  # support.
+  supports-cdi = lib.versionAtLeast version "4";
```

Comment:
This may be a singularity or an apptainer `version`

Reply:
Not familiar with Apptainer/Singularity, as far as I have checked, Apptainer does not support CDI, right?

Reply:
I don't think it does (yet). Shouldn't matter because we can keep the unconditional nvccli flag?